tests/test_mpirun.py

```diff
@@ -10,16 +10,17 @@
 
 
 @pytest.mark.skipif(sys.platform.startswith("darwin"), reason="skipping macOS test as problem with file in pytest")
-def test_mpi_run(tmpdir):
+@pytest.mark.parametrize('repeatdt', [200*86400, 10*86400])
+@pytest.mark.parametrize('maxage', [600*86400, 10*86400])
+def test_mpi_run(tmpdir, repeatdt, maxage):
     if MPI:
-        repeatdt = 200*86400
         stommel_file = path.join(path.dirname(__file__), '..', 'parcels',
                                  'examples', 'example_stommel.py')
         outputMPI = tmpdir.join('StommelMPI.nc')
         outputNoMPI = tmpdir.join('StommelNoMPI.nc')
 
-        system('mpirun -np 2 python %s -p 4 -o %s -r %d' % (stommel_file, outputMPI, repeatdt))
-        system('python %s -p 4 -o %s -r %d' % (stommel_file, outputNoMPI, repeatdt))
+        system('mpirun -np 2 python %s -p 4 -o %s -r %d -a %d' % (stommel_file, outputMPI, repeatdt, maxage))
+        system('python %s -p 4 -o %s -r %d -a %d' % (stommel_file, outputNoMPI, repeatdt, maxage))
```

Comment:
well, with respect to what I showed with pyccel, this isn't great testing. The issue here is that it doesn't really check the correct processing of that MPI call. `system` just starts the python program an returns the return value of the `__main__` function. Hard to trace back the error here. Even more so, if using the `system(...)` system call, then we should at least evaluate the return value of the function and assert that that value is 0.

Alternatively, we could let the mpi-command be run by the ci-workflow, as *pyccel* is doing it.

If uou generally agree, but rather like to move it to a new issue, then let's do that - equally valid, I could understand that argumentation.

Reply:
Yes @ckehl, I think that more careful MPI testing should be an Issue/development on its own, potentially together with a refactoring of the MPI functionality in Parcels?